branchff

```diff
@@ -0,0 +1,176 @@
+#!/bin/bash
+#
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Set PROGram name
+PROG=${0##*/}
+########################################################################
+#+
+#+ NAME
+#+     $PROG - Fast forward a branch to master via merge
+#+
+#+ SYNOPSIS
+#+     $PROG  [--nomock] <branch> [master object]
+#+     $PROG  [--helpshort|--usage|-?]
+#+     $PROG  [--help|-man]
+#+
+#+ DESCRIPTION
+#+     $PROG fast forwards a K8s release branch to [master object] (defaults to
+#+     HEAD and then prepares the branch as a K8s release branch:
+#+     * build/versionize-docs.sh <branch>
+#+     * Run hack/update-all.sh to ensure compliance of generated files
+#+
+#+
+#+ OPTIONS
+#+     [--nomock]                 - Actually force push this upstream
```

Comment:
The script currently runs in --dry-run mode by default and also prompts prior to the push (--dryrun or not).  I would be happy to add more checks if you can suggest any.